prereise/gather/hiflddata/data_trans.py

```diff
@@ -0,0 +1,645 @@
+
+    :param networkx.Graph G: Graph generated by :func:`GraphOfNet`
+    :return: (*list*) -- list of nodes inside the largest island.
+    """
+
+    Max_nodeSet = []
+    Max_size = 0
+    # list1 = []
+    num = 0
+    for c in nx.connected_components(G):
+        num = num + 1
+        nodeSet = G.subgraph(c).nodes()
+        size = len(nodeSet)
+        # list1.append(size)
+        if size > Max_size:
+            Max_nodeSet = nodeSet
+            Max_size = size
+    # list1.sort(reverse=True)
+    return Max_nodeSet
+
+
+def InitKV(clean_data):
+    """Calculate the base_KV for each node based on MIN_VOLT and MAX_VOLT;
+    Save the invalid substations as to be calculated later
+
+    :param pandas.DataFrame clean_data: substation dataframe as returned by :func:`Clean`
+    :return: (*dict*) -- KV_dict, a dict of substation's baseKV value.
+    :return: (*list*) -- to_cal, a list of substations to be calculated using neighbour search.
+    """
+
+    KV_dict = {}
+    to_cal = []
+    Max_Value = 3000
+    Min_Value = 0
+    for index, row in clean_data.iterrows():
+        base_KV = default_base_kv
+        Min_Vol = row["MIN_VOLT"]
+        Max_Vol = row["MAX_VOLT"]
+        if Min_Vol >= Max_Value or Min_Vol <= Min_Value:
+            if Max_Vol < Max_Value and Max_Vol > Min_Value:
+                base_KV = Max_Vol
+            else:
+                to_cal.append((row["LATITUDE"], row["LONGITUDE"]))
+                continue
+        else:
+            if Max_Vol < Max_Value and Max_Vol > Min_Value:
+                base_KV = (Max_Vol + Min_Vol) / 2
+            else:
+                base_KV = Min_Vol
+        KV_dict[(row["LATITUDE"], row["LONGITUDE"])] = base_KV
+    return KV_dict, to_cal
+
+
+# Give a node, find its neighbors in 5 iteration
+
+
+def get_neigbors(g, node, depth=1):
+    """Give a node, find its neighbors in given number depth
+
+    :param networkx.Graph g: Graph generated by :func:`GraphOfNet`
+    :param tuple node: substation coordinate tuple
+    :param int depth: number of hops that we are searching the neighbor substations
+    :return: (*list*) --  a list of closest substations.
+    """
+    output = {}
+    layers = dict(nx.bfs_successors(g, source=node, depth_limit=depth))
+    nodes = [node]
+    for i in range(1, depth + 1):
+        output[i] = []
+        for x in nodes:
+            output[i].extend(layers.get(x, []))
+        nodes = output[i]
+    return output
+
+
+def Cal_KV(N_dict, G, KV_dict, to_cal):
+    """Estimate missing substation's KV data using the neighbors average calculation
+
+    :param dict N_dict: dict of substation's neighbors as returned by :func:`Neighbors`
+    :param networkx.Graph g: Graph generated by :func:`GraphOfNet`
+    :param dict KV_dict: dict of existing baseKV for each substation
+    :param list to_cal: list of the substations to be calculate using average KV
+    :return: (*dict*) --  dict of all substation's baseKV being corrected and calculated.
+    """
```

Comment:
It seems that this function modifies KV_dict inplace and also returns it? What is the point of doing both?

Reply:
you are right, no need to return it. will remove the `return` in the next revision.